### PR TITLE
More implicit array to bool conversions

### PR DESCRIPTION
**Pull request by yebblies**
Another one for https://github.com/D-Programming-Language/dmd/pull/2885

### CODE REVIEW

**Comment by schveiguy** [comment]
Missing a ton of semi-colons :)

**Comment by yebblies** [comment]
haha so it is.

**Comment by yebblies** [comment]
That looks better.

**Comment by schveiguy** [comment]
Ugh, this change makes me lament we cannot just change the if(arr) to mean if(arr.length). The if(auto x = ...) syntax is so much nicer.

File: src/rt/sections_osx.d [hunk 1]
```diff
@@ -206,31 +208,34 @@ extern (C) void sections_osx_onAddImage(in mach_header* h, intptr_t slide)
...
+        auto p = cast(immutable(FuncTable)*)ehsect.ptr;
+        immutable len = ehsect.length / (*p).sizeof;
 
         _sections._ehTables = p[0 .. len];
     }
 
-    if (auto sect = getSection(h, slide, "__DATA", "__tls_data"))
+    auto tlssect = getSection(h, slide, "__DATA", "__tls_data");
+    if (tlssect)
     {
```
Comment (MartinNowak) [inline src/rt/sections_osx.d]: != null

File: src/rt/sections_osx.d [hunk 2]
```diff
@@ -206,31 +208,34 @@ extern (C) void sections_osx_onAddImage(in mach_header* h, intptr_t slide)
...
-        debug(PRINTF) printf("  tls_data %p %p\n", sect.ptr, sect.ptr + sect.length);
-        _sections._tlsImage[0] = (cast(immutable(void)*)sect.ptr)[0 .. sect.length];
+        debug(PRINTF) printf("  tls_data %p %p\n", tlssect.ptr, tlssect.ptr + tlssect.length);
+        _sections._tlsImage[0] = (cast(immutable(void)*)tlssect.ptr)[0 .. tlssect.length];
     }
 
-    if (auto sect = getSection(h, slide, "__DATA", "__tlscoal_nt"))
+    auto tlssect2 = getSection(h, slide, "__DATA", "__tlscoal_nt");
+    if (tlssect2)
```
Comment (MartinNowak) [inline src/rt/sections_osx.d]: != null

**Comment by yebblies** [comment]
@MartinNowak Oops, fixed.

**Comment by MartinNowak** [comment]
Auto-merge toggled on

**Pull request merged**